### DISCUSSION TITLE
tB IDE feature requests

### DISCUSSION THREAD

**Discussion by YogiYang** [twinbasic/twinbasic#461] [Ideas]
The IDE is coming up great!! I hope you will add support for Locking controls so that they cannot be moved or resized at design time. 

Also add support for grouping controls. Of course we can group them at design time by placing all controls that we want to group in a Frame or some other container control. 

But instead of that, giving support for grouping discreet controls will help a lot. Like for example I have 50+ controls on a form and I want to hide/show or enable/disable a few controls based on some values selected by user and the controls are NOT placed together at one location or in one container. And in my case such a situation occurs frequently.

Also allow control to be part of multiple groups. We cannot have something like this using container controls!

What developer can do is selected the controls and group them and s/he has to name the group. So s/he can show/hide or enable/disable controls by using the group name.

I have observed that enabling and disabling container controls does not change the look of the contained controls to make them look disabled and this tends to confuse users of the software

**Comment by dmrvb** [comment]
You should be able to solve your grouping requirement using existing VB functionality.
Every control has a "tag" property so this could be used to indicate which group a control belongs to.
Another approach would be to put the controls into various collections and then iterate through each collection.
Once we have full user control of the appearance of controls you could maybe do all sorts of things to emphasise if controls are enabled or even to highlight which grouping they belong to.

**Reply by wqweto** [reply]
OP wants to mass manipulate design-time properties on groups of controls e.g. change the font of all textboxes at design-time or bring to front a group of controls.

**Reply by YogiYang** [reply]
Not only at design time but at runtime also.
